### ISSUE TITLE
`no-useless-template` rule

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-unicorn#71]
I would like to prevent use of template literals when not needed. It's only needed when it contains an expansion `${}` or literal newlines.

We could also consider supporting the case of using template literal to reduce escaping when you for example enforce single-quotes, but the string also contains single-quotes. This would have to be an option where to user specifics the preferred quote type as we're not allowed to read it from the `quotes` rule.

Should be fixable.

Thoughts? *(Anyone)*

---

Example of when it would have been useful: https://github.com/sindresorhus/eslint-plugin-unicorn/pull/67#discussion_r85975675

**Comment by sindresorhus** [comment]
There's the built-in [`quotes`](http://eslint.org/docs/rules/quotes) rule which can enforce a particular quote, but it only has an option to *always* allow `allowTemplateLiterals`, which I use, but I would like to only allow it when needed. Maybe worth opening an ESLint issue about adding an option for that?

**Comment by jfmengels** [comment]
This would be much better in the built-in `quotes` rule IMO. I'm :+1: for opening an issue over there instead.

**Comment by jfmengels** [comment]
This should probably also not be enforced when there are line breaks in the template literal, like here:
```js
`foo
bar`
```

**Comment by sindresorhus** [comment]
> This would be much better in the built-in quotes rule IMO. I'm 👍 for opening an issue over there instead.

I don't have time to deal with the process of suggesting a new option on the ESLint tracker (it's a lot of work and bikeshedding). Would appreciate if anyone else could do it. See: http://eslint.org/docs/developer-guide/contributing/rule-changes

**Comment by SamVerschueren** [comment]
What's up with this one? Are we going to try to add it to the `quotes` rule or are we going to implement it ourselves? 

@jfmengels Did you already create an issue for this? If so, it's probably already closed but just asking :).

**Comment by sindresorhus** [comment]
Let's just do it as a new rule.

**Comment by MrHen** [comment]
I'm not sure what the difference between this and the [quotes](https://eslint.org/docs/rules/quotes) rule would be. Maybe the quotes rule was updated since the original discussion?

Rule:
```
      "quotes": [
        "error",
        "single"
      ]
```

Pass:
```
export const foo = 'foo';
export const bar = `${foo}`;
```

Fail:
```
export const zaz = `foo`;
```

@sindresorhus can you clarify what pass/fail conditions you expect?

**Comment by sindresorhus** [comment]
@MrHen You're right. That's great! It must have changed since I opened this issue.
